### PR TITLE
Use disk stats, not RAM, for shrink action and node evaluation.

### PR DESCRIPTION
**Pull request by linuxboyng**
#417

### CODE REVIEW

**Comment by downsrob** [comment]
Our developer certificate of origin check is failing, would you mind signing off your commit and force pushing?
You can sign it with:
```
git commit --amend --signoff
```
See here https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#developer-certificate-of-origin for more information

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/shrink/AttemptMoveShardsStep.kt [hunk 1]
```diff
@@ -394,6 +394,7 @@ class AttemptMoveShardsStep(private val action: ShrinkAction) : ShrinkStep(name,
 
     companion object {
         const val OS_METRIC = "os"
```
Comment (downsrob) [inline src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/shrink/AttemptMoveShardsStep.kt:396]: Can you change line 270 to FS_METRIC, and remove this constant?

File: src/test/kotlin/org/opensearch/indexmanagement/indexstatemanagement/util/StepUtilsTests.kt [hunk 2]
```diff
@@ -19,7 +19,8 @@ import org.opensearch.indexmanagement.spi.indexstatemanagement.model.ActionMetaD
 import org.opensearch.indexmanagement.spi.indexstatemanagement.model.ManagedIndexMetaData
 import org.opensearch.indexmanagement.spi.indexstatemanagement.model.ShrinkActionProperties
 import org.opensearch.jobscheduler.spi.LockModel
-import org.opensearch.monitor.os.OsStats
+// import org.opensearch.monitor.os.OsStats
```
Comment (downsrob) [inline src/test/kotlin/org/opensearch/indexmanagement/indexstatemanagement/util/StepUtilsTests.kt:22]: You can remove this commented out import as well, please.

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/shrink/AttemptShrinkStep.kt [hunk 3]
```diff
@@ -60,8 +60,8 @@ class AttemptShrinkStep(private val action: ShrinkAction) : ShrinkStep(name, tru
             return false
         }
         val indexSizeInBytes = statsStore.sizeInBytes
-        // Get the remaining memory in the node
-        val nodesStatsReq = NodesStatsRequest().addMetric(AttemptMoveShardsStep.OS_METRIC)
+        // Get the remaining memory and disk space in the node.
+        val nodesStatsReq = NodesStatsRequest().addMetrics(AttemptMoveShardsStep.OS_METRIC, AttemptMoveShardsStep.FS_METRIC)
```
Comment (downsrob) [inline src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/shrink/AttemptShrinkStep.kt:64]: You can remove the OS_METRIC entirely, we aren't using it at all anymore

**Review by downsrob — CHANGES_REQUESTED** [review]

**Comment by linuxboyng** [comment]
Making those suggested changes now.

**Comment by linuxboyng** [comment]
Pushed changes to my fork on same branch.

**Comment by downsrob** [comment]
Thanks! Code changes look great. Last thing is that your commits still need to be signed off.
If your user.name and user.email are set in your git configs you can use this to sign off your last two commits
```
git rebase --signoff HEAD~2
```
Then just force push the change. 
Otherwise, you can just add `Signed-off-by: Your Name <your.email@email.com>` to the end of each of your commit messages directly. By the way, you will need to use your real name, we don't accept pseudonyms or anonymous contributors.

**Comment by downsrob** [comment]
No worries. We are almost there, but I see that your first two commits are still failing DCO check as those aren't signed off. Can you amend these two: 3afe1b3, ac6959c and force push?

**Comment by linuxboyng** [comment]
Sorry but my git knowledge is limited.  How would I amend those two specific commits?  I tried doing a `git checkout ac6959c` then `git commit --amend --signoff` but it keeps getting rejected with the error `Updates were rejected because a pushed branch tip is behind its remote counterpart` .  Would it be easier to just redo the entire process again?

**Comment by downsrob** [comment]
To sign all of your commits (the last 6) you can use this:
```
git rebase HEAD~6 --signoff
```
Your changes are rewriting the commit history, so you will need to force push. You can just use this:
```
git push --force
```

**Comment by linuxboyng** [comment]
Done.  Ty again!

**Comment by downsrob** [comment]
Ok, so it looks like you removed some of your commits so when you signed off the batch of previous commits, you ended up modifying two existing commits from main. Now those commits are being included in this PR. We don't want to rewrite those changes so we will have to undo that. There are a lot of ways to fix this but the easiest one will be to just reset the commit history, add a new commit, and then push.
Here is what I will recommend:
First just make a backup branch in case we break everything here, we don't want to lose the code. Then go back to this branch.
```
git checkout use_diskstats_for_shrink
git checkout -b backup-shrink-fix
git checkout use_diskstats_for_shrink
```
Reset your git history to main, this will keep your file changes but reset the git history.
```
git reset --soft origin/main
```
Then just create a new commit with your changes:
```
git commit -m "Use disk stats, not RAM, for shrink action and node evaluation." --signoff
```
Now force push as you will be modifying the history of the pull request:
```
git push --force
```

**Comment by linuxboyng** [comment]
I ran your suggested git commands.  Hopefully I did it correct this time.

**Review by downsrob — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]
Thanks for the change!

**Pull request merged**